clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy

```diff
@@ -142,6 +142,9 @@ class ServerGroupController {
         sg.instanceType = serverGroup.launchConfig.instanceType
       }
     }
+    if (serverGroup.launchTemplate && serverGroup.launchTemplate.launchTemplateData) {
```

Comment:
Should this be an `else if` with the block above? Are launch config and launch templates mutually exclusive?

Reply:
Yes, the serverGroup will have either a launch config or a launch template, but never both.

Reply:
I believe we could add a `getInstanceType()` to `AmazonServerGroup` and make the determination there.

As it stands now, this is referencing an attribute `launchTemplate` that is not defined on the `ServerGroup` interface. Of course because it's groovy, it'll work for AmazonServerGroup but not for any other providers.